### PR TITLE
0.11.0 Migration guide

### PR DESCRIPTION
**Pull request by war-in**
### **Please check if the PR fulfills these requirements**
- [ ] The formatter, linter, and tests all run without an error
- [ ] Tests for the changes have been added (for bug fixes/features)
- [ ] Docs have been added / updated (for bug fixes / features)

## **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)

Introduces `Migration guide` for the new version.
It has to be modified if the `development` will be updated

## **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)

No

### CODE REVIEW

File: docs/migration_guide.rst [hunk 1]
```diff
@@ -1,11 +1,30 @@
+0.11.0 Migration guide
+======================
+
+Cairo-lang 0.10.3 dropped support for the `Deploy` transaction. To be compatible we had to remove some
+(already deprecated) features.
+
+Breaking Changes
+----------------
+
+Methods below are removed:
```
Comment (cptartur) [inline docs/migration_guide.rst:10]:
```suggestion
Removed methods:
```

File: docs/migration_guide.rst [hunk 2]
```diff
@@ -1,11 +1,30 @@
...
+
+- deprecated `Contract.deploy`. Read more about deployment in the `Deploying contracts <https://starknetpy.readthedocs.io/en/latest/guide.html#deploying-contracts>`_ section.
+- deprecated `AccountClient.create_account`. `Account creation <https://starknetpy.readthedocs.io/en/latest/account_creation.html>`_ docs are here to help you!
+- `Client.deploy` method (from the interface and all implementations)
+- deprecated `make_deploy_tx`
+- `compute_deploy_hash`
+- the `Deploy` transaction
```
Comment (cptartur) [inline docs/migration_guide.rst:17]: I don't think we have to emphasise these were deprecated, but it's up to you.

File: docs/migration_guide.rst [hunk 3]
```diff
@@ -1,11 +1,36 @@
+0.11.0 Migration guide
+======================
+
+Cairo-lang 0.10.3 dropped support for the `Deploy` transaction. To be compatible we had to remove some
+(already deprecated) features.
```
Comment (Solpatium) [inline docs/migration_guide.rst:5]: just "some deprecated features."

File: docs/migration_guide.rst [hunk 4]
```diff
@@ -1,11 +1,36 @@
...
+- `Client.deploy` method (from the interface and all implementations)
+- `make_deploy_tx`
+- `compute_deploy_hash`
+- the `Deploy` transaction
+
+
+Invoke Transaction
+------------------
+
+Old `InvokeFunction` transaction is now aliased as `Invoke`. We suggest to start using the new `Invoke`.
```
Comment (Solpatium) [inline docs/migration_guide.rst:23]: I suggest moving it above breaking changes.

File: docs/migration_guide.rst [hunk 5]
```diff
@@ -1,11 +1,36 @@
+0.11.0 Migration guide
+======================
+
+Cairo-lang 0.10.3 dropped support for the `Deploy` transaction. To be compatible we had to remove some
+(already deprecated) features.
+
+Breaking Changes
+----------------
+
+Removed methods:
```
Comment (Solpatium) [inline docs/migration_guide.rst:10]: Removed API? `Deploy` is not a method

**Review by Solpatium — APPROVED** [review]
LGTM with comments

File: docs/migration_guide.rst (hunk 4 above)
Comment (war-in) [reply docs/migration_guide.rst:23]: See your point, but `Breaking changes` are always on the top

Comment (cptartur) [reply docs/migration_guide.rst:23]: I'd keep breaking changes where they are, it's the most important thing from the user perspective when they check migration guide.

**Review by cptartur — APPROVED** [review]
After all existing discussions are addressed, can be merged.

**Pull request merged**